### DISCUSSION TITLE
Can Web3Modal be integrated into a chrome extension?

### DISCUSSION THREAD

**Discussion by javaspeak** [WalletConnect/walletconnect-monorepo#2459] [Developer Support] [answered]
Hi can Web3Modal be integrated into a chrome extension?

The reason I am asking this is, possibly Web3Modal is attaching contexts to the window of the host website and possibly there is no mechanism to access this context from within the chrome extension?  Anyone Know of an example which shows integrating latest version of web3modal into a chrome extension manifest v3 ?

**Comment by javaspeak** [comment]
Anyone?

**Comment by blaisebakundukize** [comment]
Me too I want to know the answer!

**Comment by finessevanes** [comment, marked as answer]
In theory, Web3Modal can be integrated into a Chrome extension. However, there might be challenges when it comes to interacting with injected wallets. We are not completely certain whether inter-extension communication of this nature is feasible.

**Reply by javaspeak** [reply]
Thank you for your answer.

Meta Mask has a in page provider option which communicates via a port.  This allows an extension to communicate with a wallet extension without hacking by injecting scripts in the host dom.  The problem of injecting scripts is that approach does not work if the host website sets up a security policy which disallows script injection.

I would like to somehow know how many wallets are copying Meta Mask functionality and have a provider which can connect over ports instead of communicating by injecting stuff like window.ethereum.

The problem of extensions is that they cannot see any attributes that were injected in the host window context (when inside the extension window context).
